ocaml/xapi/xapi_pvs_farm.ml

```diff
@@ -26,33 +26,108 @@ let introduce ~__context ~name =
 		~ref:pvs_farm ~uuid ~name ~cache_storage:[];
 	pvs_farm
 
+(** [proxies] returns all currently attached proxies *)
+let proxies ~__context ~self =
+	let open Db_filter_types in
+	Db.PVS_proxy.get_refs_where ~__context
+		~expr:
+			(And
+				((Eq (Field "farm", Literal (Ref.string_of self)))
+				,(Eq (Field "currently_attached", Literal "true"))
+				))
+
 let forget ~__context ~self =
 	let open Db_filter_types in
 	(* Check there are no running proxies. *)
-	let running_proxies = Db.PVS_proxy.get_refs_where ~__context
-		~expr:(And
-			((Eq (Field "farm", Literal (Ref.string_of self))),
-			(Eq (Field "currently_attached", Literal "true"))))
-	in
-	if running_proxies <> []
-	then raise Api_errors.(Server_error
-		(pvs_farm_contains_running_proxies,
-		List.map Ref.string_of running_proxies));
+	let running_proxies = proxies ~__context ~self in
+	if running_proxies <> [] then
+		raise Api_errors.(Server_error
+			(pvs_farm_contains_running_proxies,
+			List.map Ref.string_of running_proxies));
 	(* Check there are no servers. *)
 	let servers = Db.PVS_farm.get_servers ~__context ~self in
-	if servers <> []
-	then raise Api_errors.(Server_error
+	if servers <> [] then raise Api_errors.(Server_error
 		(pvs_farm_contains_servers, List.map Ref.string_of servers));
 	Db.PVS_farm.destroy ~__context ~self
 
+(** set the name of [self] *)
 let set_name ~__context ~self ~value =
-  not_implemented "PVS_farm.set_name"
+	if proxies ~__context ~self <> [] then
+		not_allowed "can't set a name while proxies are still active"
+	else
+		Db.PVS_farm.set_name ~__context ~self ~value
+
+(** [disjoint] is [true] if two lists don't share at least one element *)
+let disjoint xs ys =
+	List.for_all (fun x -> not @@ List.mem x ys) xs
```

Comment:
This would be nice to put in https://github.com/xapi-project/stdext/blob/master/lib/listext.ml